src/utils/location.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import ScenarioMarker from "../telemetry/ScenarioMarker";
+import TelemetryEvent from "../telemetry/TelemetryEvent";
+
+interface Location {
+    latitude: string;
+    longitude: string;
+}
+
+export const getLocationInfo = (scenarioMarker: ScenarioMarker, chatId: string, requestId: string): Promise<Location> => {
+    const reportLocationError = (response: string, error?: any) => {
+        console.error(response, error);
+        let exceptionDetails;
+        try {
+            exceptionDetails = JSON.stringify(error)
+        } catch {
+            exceptionDetails = JSON.stringify({Response: response})
+        }
+        scenarioMarker.failScenario(TelemetryEvent.GetGeoLocation, {
+            RequestId: requestId,
+            ChatId: chatId,
+            ExceptionDetails: exceptionDetails
+        });
+    }
+
+    return new Promise((resolve, reject) => {
```

Comment:
Please remove `new Promise()` since we'll add `async` on the method instead. It will be a promise by default in this situation